src/Database/DBA.php

```diff
@@ -850,411 +460,39 @@ public static function unlock() {
-						}
-						return false;
-					}
-				}
-			}
-		}
-		if ($do_transaction) {
-			self::commit();
-		}
-		return true;
+		return self::$db->delete($table, $conditions, (isset($options['cascade']) ? $options['cascade'] : true));
 	}
 
-	/**
-	 * @brief Updates rows
-	 *
-	 * Updates rows in the database. When $old_fields is set to an array,
-	 * the system will only do an update if the fields in that array changed.
-	 *
-	 * Attention:
-	 * Only the values in $old_fields are compared.
-	 * This is an intentional behaviour.
-	 *
-	 * Example:
-	 * We include the timestamp field in $fields but not in $old_fields.
-	 * Then the row will only get the new timestamp when the other fields had changed.
-	 *
-	 * When $old_fields is set to a boolean value the system will do this compare itself.
-	 * When $old_fields is set to "true" the system will do an insert if the row doesn't exists.
-	 *
-	 * Attention:
-	 * Only set $old_fields to a boolean value when you are sure that you will update a single row.
-	 * When you set $old_fields to "true" then $fields must contain all relevant fields!
-	 *
-	 * @param string        $table      Table name
-	 * @param array         $fields     contains the fields that are updated
-	 * @param array         $condition  condition array with the key values
-	 * @param array|boolean $old_fields array with the old field values that are about to be replaced (true = update on duplicate)
-	 *
-	 * @return boolean was the update successfull?
-	 * @throws \Exception
-	 */
-	public static function update($table, $fields, $condition, $old_fields = []) {
-
-		if (empty($table) || empty($fields) || empty($condition)) {
-			Logger::log('Table, fields and condition have to be set');
-			return false;
-		}
-
-		$table = self::escape($table);
-
-		$condition_string = self::buildCondition($condition);
-
-		if (is_bool($old_fields)) {
-			$do_insert = $old_fields;
-
-			$old_fields = self::selectFirst($table, [], $condition);
-
-			if (is_bool($old_fields)) {
-				if ($do_insert) {
-					$values = array_merge($condition, $fields);
-					return self::insert($table, $values, $do_insert);
-				}
-				$old_fields = [];
-			}
-		}
-
-		$do_update = (count($old_fields) == 0);
-
-		foreach ($old_fields AS $fieldname => $content) {
-			if (isset($fields[$fieldname])) {
-				if (($fields[$fieldname] == $content) && !is_null($content)) {
-					unset($fields[$fieldname]);
-				} else {
-					$do_update = true;
-				}
-			}
-		}
-
-		if (!$do_update || (count($fields) == 0)) {
-			return true;
-		}
-
-		$sql = "UPDATE `".$table."` SET `".
-			implode("` = ?, `", array_keys($fields))."` = ?".$condition_string;
-
-		$params1 = array_values($fields);
-		$params2 = array_values($condition);
-		$params = array_merge_recursive($params1, $params2);
-
-		return self::e($sql, $params);
+	public static function update($table, $fields, $condition, $old_fields = [])
+	{
+		return self::$db->select($table, $fields, $condition, $old_fields);
 	}
 
-	/**
```

Comment:
Please don't remove or relocate this block. This is - more or less - the documentation for the DBA functions, no matter if they are for MySQLi or PDO or what ever.

Reply:
Since the `DBA` method will become a static wrapper for the `Database` class, isn't it enough to have these comments on the `Database` class methods?

Reply:
The DBA class is used by the coders to work in the core. Advanced text editors can display these comments when writing code that uses these DBA methods.

Reply:
The `DBA` class should/will at least be replaced by the `Database` class inside the `Core` namespace (like we've done it for Configuration, App\Mode, App\Router, Logger). In fact this will reduce massive the chaining between the classes.

Maybe we can use the `@see IDatbase::*` annotation for further description, so they can be used at both sides? http://docs.phpdoc.org/references/phpdoc/tags/see.html

As said, the main difference for a code between the two paradigmas is:
```php
use Friendica\Database\DBA;

class myClass
{
  public function aMethod()
  {
      DBA::select();
  }
}
```

```php
use Friendica\Database\IDatabase;

class myClass
{
  /**
   * @var IDatabase
   **/
  private $dba;
  
  public function __construct(IDatabase $dba)
  {
     $this->dba = $dba;
  }
  
  public function aMethod()
  {
     $dba = $this->dba;
     $dba->select();
  }
}
```

So the main difference is that a coder has to construct a class with its depencies. There, the future DI-framework will get responsible to automatically construct this dependencies. So it should be a small effort to use them for new coders.

Reply:
Sorry, but I really dislike that change. When we - from now on - have to do this whole "construct"-thingy, this only makes it more complicated. From a pragmatic programmer's point of view I don't see any advantage to the current way.

Reply:
The pragmatic advantage is that there is no global magic, you know were the DB capability is coming from because it was passed in the class constructor. This class constructor can in turn be traced back, and the original DB object instantiation can be found there.

It feels more complicated because you know where the static `DBA` class is initialized from experience, but if you just look at a class without knowing anything about the project, it just is some global magic you can either go with or lose some time finding out where and when it is initialized, and then it become the experience you have.

Furthermore, it is easier to test if you can provide a Database object you control. Otherwise, you're testing both the current class and whatever global static classes is called. This in turn increases test duration and makes me forget PRs that I approved but took too long to finish the tests.